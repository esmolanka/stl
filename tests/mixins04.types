
type Nested = mixin f
  { foo : f Int
  , bar : { baz : f Int, quux : f Float }
  }
#eval Nested

type NestedRecord = { mix Nested }
#eval NestedRecord

type Array (n : Nat) a = a[n]
type NestedTable = exists (n : Nat). record of (Array n) { mix Nested }
#eval NestedTable
