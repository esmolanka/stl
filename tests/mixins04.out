tests/mixins04.types:6:1-6:13:
  Normalised:
    λ (f : Type → Type). λ (ρ : Row).
      (foo : f Int, bar : ∃ (ρ : Row). {baz : f Int, quux : f Float, … ρ}, … ρ)
  Kind:
    (Type → Type) → Row → Row
tests/mixins04.types:9:1-9:19:
  Normalised:
    ∃ (ρ : Row).
      {foo : Int, bar : ∃ (ρ : Row). {baz : Int, quux : Float, … ρ}, … ρ}
  Kind:
    Type
tests/mixins04.types:13:1-13:18:
  Normalised:
    ∃ (n : Nat). ∃ (ρ : Row).
      { foo : Int[n]
      , bar : ∃ (ρ : Row). {baz : Int[n], quux : Float[n], … ρ}
      , … ρ
      }
  Kind:
    Type